packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -585,6 +647,61 @@ abstract class RenderSliverFloatingPersistentHeader extends RenderSliverPersiste
     _lastActualScrollOffset = constraints.scrollOffset;
   }
 
+  @override
+  void showOnScreen({
+    RenderObject descendant,
+    Rect rect,
+    Duration duration = Duration.zero,
+    Curve curve = Curves.ease,
+  }) {
+    assert(child != null);
+    final Rect childBounds = descendant != null
+      ? MatrixUtils.transformRect(descendant.getTransformTo(child), rect ?? descendant.paintBounds)
+      : null;
+
+    double minTargetExtent;
+    switch (applyGrowthDirectionToAxisDirection(constraints.axisDirection, constraints.growthDirection)) {
+      case AxisDirection.up:
+        minTargetExtent = childExtent - (childBounds?.top ?? 0);
+        break;
+      case AxisDirection.right:
+        minTargetExtent = childBounds?.right ?? childExtent;
+        break;
+      case AxisDirection.down:
+        minTargetExtent = childBounds?.bottom ?? childExtent;
+        break;
+      case AxisDirection.left:
+        minTargetExtent = childExtent - (childBounds?.left ?? 0);
+        break;
+    }
+
+    minTargetExtent = minTargetExtent.clamp(childExtent, maxExtent) as double;
+    // Expands the header if needed, with animation if possible.
+    if (minTargetExtent > childExtent) {
```

Comment:
Maybe I missed it, but did you have tests for when the header animates and for when it doesn't?